nautobot_ssot/integrations/vsphere/utilities/vsphere_client.py

```diff
@@ -0,0 +1,169 @@
+"""Extending vSphere SDK."""
+
+import logging
+import re
+import urllib.parse
+from dataclasses import dataclass
+from typing import Dict
+
+import requests
+import urllib3
+from requests.auth import HTTPBasicAuth
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_url(address):
+    """Handle outside case where protocol isn't included in URL address.
+
+    Args:
+        address (str): URL set by end user for Infoblox instance.
+
+    Returns:
+        ParseResult: The parsed results from urllib.
+    """
+    if not re.search(r"^[A-Za-z0-9+.\-]+://", address):
+        address = f"https://{address}"
+    return urllib.parse.urlparse(address)
+
+
+class InvalidUrlScheme(Exception):
```

Comment:
This is a duplicate of the Exception in the project root folder. Please update to use that one.